central/sensor/service/pipeline/nodes/pipeline.go

```diff
@@ -90,13 +90,24 @@ func (p *pipelineImpl) Run(ctx context.Context, clusterID string, msg *central.M
 		node.ClusterName = clusterName
 	}
 
+	if enricher.SupportsNodeScanning(node) {
+		// Call upsert without any components, only Node metadata. Upsert will read any
+		// components from the database before writing. Safe because NodeInventory and
+		// Node pipelines never run concurrently by the Sensor Event worker queues.
+		if err := p.nodeDatastore.UpsertNode(ctx, node); err != nil {
```

Comment:
I want to confirm my thought about the `if` block:
You are skipping the enrichment and analysis for host-scannable nodes because the inventory pipeline is guaranteed to provide more accurate vuln insights.

If the above is true, it would be a good idea to include it in the comment.

Reply:
Additionally, I think you mentioned that you need to handle `SensorEvent_Node` for metadata but don't need to populate the `.scan` from this pipeline. How about setting it .`.scan` to `nil` here? That should guarantee the scan from this pipeline would never disturb scan in db because of the check `scan == nil` in store.

Reply:
> Maybe. I don't think there is ambiguity here regarding the role of the two pipelines for host-scannable nodes, are you feeling strong about this?

It is a change in behavior with the launch of this feature, and not many people are aware about it. It is just to save time explaining to individual basis.

> .scan is already always nil. Or, do you mean explicitly setting it to nil, to guard the beavior? That is a good idea. Let me know if you meant something else.

Correct, I meant extra safety valve.